core/genesis.go

```diff
@@ -547,12 +589,8 @@ func (g *Genesis) Commit(db ethdb.Database, triedb *trie.Database) (*types.Block
 
 // MustCommit writes the genesis block and state to db, panicking on error.
 // The block is committed as the canonical head block.
-// Note the state changes will be committed in hash-based scheme, use Commit
-// if path-scheme is preferred.
-func (g *Genesis) MustCommit(db ethdb.Database) *types.Block {
-	block, err := g.Commit(db, trie.NewDatabaseWithConfig(db, &trie.Config{
-		Zktrie: g.Config != nil && g.Config.Zktrie,
-	}))
+func (g *Genesis) MustCommit(db ethdb.Database, triedb *trie.Database) *types.Block {
+	block, err := g.Commit(db, triedb)
```

Comment:
It seems like code that communicates whether ZK or not is needed for DB update.
```suggestion
block, err := g.Commit(db, trie.NewDatabaseWithConfig(db, &trie.Config{
		Zktrie: g.Config != nil && g.Config.Zktrie,
	}))
```

Reply:
We'll modify the mustcommit function to set the zktrie field.
However, later we'll change it so that all zktrie settings must be set externally.